src/components/course/enrollment/tests/EnrollAction.test.jsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import { screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import { AppContext } from '@edx/frontend-platform/react';
+
+import { ENROLL_MODAL_TEXT_NO_OFFERS, createUseVoucherText } from '../../EnrollModal';
+import {
+  renderWithRouter,
+  initialAppState,
+  initialCourseState,
+  A_100_PERCENT_OFFER,
+} from '../../../../utils/tests';
+
+import { createCourseInfoUrl } from '../../data/utils';
+import { COURSE_MODES_MAP } from '../../data/constants';
+import EnrollAction from '../EnrollAction';
+import { enrollButtonTypes } from '../constants';
+import { CourseContextProvider } from '../../CourseContextProvider';
+import { UserSubsidyContext } from '../../../enterprise-user-subsidy';
+
+/**
+ * These tests verify that the correct enroll component is rendered.
+ * They do not test the enroll label since that is tested separately.
+ * We just pass a pre-rendered enroll label for these tests.
+ * See this page for scenarios:
+ * https://openedx.atlassian.net/wiki/spaces/SOL/pages/2178875970/Enroll+Button+logic+for+Enterprise+Learner+Portal
+ */
+const {
+  TO_COURSEWARE_PAGE,
+  VIEW_ON_DASHBOARD,
+  ENROLL_DISABLED,
+  TO_DATASHARING_CONSENT,
+  TO_ECOM_BASKET,
+  TO_VOUCHER_REDEEM,
+} = enrollButtonTypes;
+
+const INITIAL_APP_STATE = initialAppState({});
+const selfPacedCourseWithLicenseSubsidy = initialCourseState({});
+const verifiedTrackEnrollment = {
+  mode: COURSE_MODES_MAP.VERIFIED,
+  isActive: true,
+  courseDetails: {
+    courseId: selfPacedCourseWithLicenseSubsidy.activeCourseRun.key,
+  },
+};
+const subscriptionLicense = { uuid: 'a-license' };
+
+/**
+   * @param {object} args Arguments.
+   * @param {string} args.enrollAction
+   */
+const EnrollLabel = (props) => (
+  // eslint-disable-next-line react/prop-types
+  <div>{props.enrollLabelText}</div>
+);
+const renderEnrollAction = ({
+  enrollAction,
+  courseInitState = selfPacedCourseWithLicenseSubsidy,
+  initialUserSubsidyState = {
+    subscriptionLicense,
+    offers: {
+      offers: [],
+      offersCount: 0,
+    },
+  },
+}) => {
+  // need to use router, to render component such as react-router's <Link>
+  renderWithRouter(
+    <AppContext.Provider value={INITIAL_APP_STATE}>
+      <UserSubsidyContext.Provider value={initialUserSubsidyState}>
+        <CourseContextProvider initialState={courseInitState}>
+          {enrollAction}
+        </CourseContextProvider>
+      </UserSubsidyContext.Provider>
+    </AppContext.Provider>,
+  );
+};
+
+describe('Scenarios where user is enrolled in course', () => {
+  test(`to_courseware_page rendered with course info url for self paced course,
+      scenario 1`, () => {
```

Comment:
I think you'll need to have a comment above explaining what the scenarios are. I know what you're talking about because I read your writeup on this, but if I was new to the code, `scenario 1` would mean nothing to me.

Reply:
yeah I see what you mean. I removed the scenario words here since the test titles are explanatory, and the reference link is present for ref. ty!